v2.1/performance-benchmarking.md

```diff
@@ -0,0 +1,273 @@
+---
+title: Performance Benchmarking
+summary: Learn how to benchmark CockroachDB against TPC-C.
+toc: false
+---
+
+This page walks you through [TPC-C](http://www.tpc.org/tpcc/) performance benchmarking on CockroachDB. It measures `tpmC` (new order transactions/minute) on two TPC-C datasets:
+
+- 1,000 warehouses (for a total dataset size of 200GB) on 3 nodes
+- 10,000 warehouses (for a total dataset size of 2TB) on 30 nodes _(Coming soon)_
+
+These two points on the spectrum show how CockroachDB scales from modest-sized production workloads to larger-scale deployments.
+
+<!--This demonstrates how CockroachDB achieves high OLTP performance of over 128,000 tpmC on a TPC-C dataset over 2TB in size.-->
+
+<div id="toc"></div>
+
+## Benchmark a small cluster
+
+### Step 1. Start a cluster on Google Cloud Platform GCE
+
+Follow steps 1-7 in the [GCE tutorial to deploy a 3-node CockroachDB cluster on Google Cloud](deploy-cockroachdb-on-google-cloud-platform.html), with the following changes:
+
+- For the 3 CockroachDB nodes, use the `n1-highcpu-16` VMs with [Local SSD](https://cloud.google.com/compute/docs/disks/local-ssd).
+    For our TPC-C benchmarking, we use `n1-highcpu-16` machines. Currently, we believe this (or higher vCPU count machines) is the best configuration for CockroachDB under high traffic scenarios. We also attach a single local SSD to each virtual machine. Local SSDs are low latency disks attached to each VM, which maximizes performance. We do not recommend using network-attached block storage. We chose this configuration because it best resembles what a bare metal deployment would look like, with machines directly connected to one physical disk each.
+
+- Skip step 4, for setting up Google's manage load balancing service. Instead, reserve a fourth VM for read and write testing.
+
+{{site.data.alerts.callout_danger}}
+This configuration is intended for performance benchmarking only. For production deployments, there are other important considerations, such as ensuring that data is balanced across at least 3 availability zones for resiliency. See the [Production Checklist](recommended-production-settings.html) for more details.
+{{site.data.alerts.end}}
+
+<!-- ## Roachprod directions for performance benchmarking
+
+Use roachprod to create cluster: `roachprod create lauren-tpcc --gce-machine-type "n1-highcpu-16" --local-ssd --nodes 4`
+
+Download latest version of CockroachDB:
+
+- `roachprod run lauren-tpcc 'wget https://binaries.cockroachdb.com/cockroach-v2.0.3.linux-amd64.tgz'`
```

Comment:
Since this is a 2.1 page, this and the next roachprod step should reference the latest 2.1 alpha, I think.